lib/mayaUsd/fileio/primUpdaterManager.cpp

```diff
@@ -1198,7 +1198,19 @@ bool PrimUpdaterManager::discardEdits(const MDagPath& dagPath)
     MayaUsd::ProgressBarScope progressBar(1, "Discarding Converted Maya Data");
 
     auto usdPrim = MayaUsd::ufe::ufePathToPrim(pulledPath);
+
+#ifdef HAS_ORPHANED_NODES_MANAGER
+    auto ret = _orphanedNodesManager->isOrphaned(pulledPath)
+        ? discardOrphanedEdits(dagPath, pulledPath)
+        : discardPrimEdits(pulledPath);
+#else
+    // The following is incorrect: because of pull information in the session
+    // layer stored as overs, the usdPrim will never be invalid: a prim that
+    // exists only because of over opinions is valid, but is typeless.
+    // Therefore, the conditional will always succeed, and
+    // discardOrphanedEdits() is never called.  PPT, 30-Sep-2022.
```

Comment:
Could we check if it is typeless? Would that help?

Reply:
That feels weak...  And yet another code path to support and test.  My hope is that since the required UFE support in Maya is coming very shortly on all Maya versions where this matters (UFE v3 and up), that we simply remove this broken code path soon.  Thanks for the thought, though.